app/resources/v1/document/api.py

```diff
@@ -0,0 +1,140 @@
+import os
+from flask import (
+    request,
+    current_app
+)
+from flask_restful import Resource
+from werkzeug.utils import secure_filename
+from datetime import datetime
+from app.resources.lib import api_response
+# from app.models import Document
+from app.resources.lib.schema_utils import validate_json
+from flask_login import login_required
+
+SCHEMA_PATH = 'v1/document/'
+
+# TODO: use db object
+from collections import namedtuple
+
+Document = namedtuple(
+    "Document", [
+        "id",
+        "title",
+        "subtitle",
+        "agency",
+        "creators",
+        "report_type",
+        "subjects",
+        "date_published",
+        "year",
+        "start_date",
+        "end_date",
+        "description"
+    ]
+)
+
+
+class DocumentAPI(Resource):
+    def get(self, id):
+        doc = Document(id,
+                       'A Title',
+                       'TYPE',
+                       'A description.')
+        return api_response.success({
+            'document': {
+                'id': doc.id,
+                'title': doc.title,
+                'type': doc.type,
+                'description': doc.description,
+            }
+        })
+
+    @login_required
+    def post(self):
+        try:
+            json = request.get_json(force=True)
+            # TODO: utility for stripping values
+        except Exception as e:
```

Comment:
I've looked online and many of the use cases of `get_json` do not have a try block. In the few cases that I've seen, only the general Exception is caught. I've tried to play around with it and I've gotten AttributeError raised but there might be more specific exceptions that I don't know about.

I agree that we should setup logging early on to help with debugging.